hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -495,7 +495,7 @@ enum Value extends Path:
   case Arr(elems: Ls[Arg])
   case Rcd(elems: Ls[RcdArg])
 
-case class Arg(spread: Bool, value: Path)
+case class Arg(spread: Bool, value: Path, eager: Bool = true)
```

Comment:
Make illegal states unrepresentable
```suggestion
case class Arg(spread: Opt[Bool], value: Path)
```